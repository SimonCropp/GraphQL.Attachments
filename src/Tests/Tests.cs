using System.IO;
using System.Threading.Tasks;
using GraphQL;
using GraphQL.Attachments;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.DependencyInjection;
using ObjectApproval;
using Xunit;
using Xunit.Abstractions;

public class Tests: TestBase
{
    [Fact]
    public async Task Mutation()
    {
        var mutation = @"
mutation
{
  withAttachment (argument: 'argumentValue')
  {
    argument
  }
}";
        var result = await RunQuery(mutation);
        ObjectApprover.VerifyWithJson(result);
    }

    [Fact]
    public async Task Query()
    {
        var queryString = @"
{
  withAttachment (argument: 'argumentValue')
  {
    argument
  }
}";
        var result = await RunQuery(queryString);
        ObjectApprover.VerifyWithJson(result);
    }

    static async Task<AttachmentExecutionResult> RunQuery(string queryString)
    {
        var incomingAttachments = new IncomingAttachments();
        var memoryStream = BuildStream();
        var metadata = new HeaderDictionary
        {
            {"header1", "headerValue"}
        };
        incomingAttachments.Add("key", new AttachmentStream("key", memoryStream, 3, metadata));
        var services = new ServiceCollection();

        TestServices.AddGraphQlTestTypes(services);

        var executeQuery = await QueryExecutor.ExecuteQuery(queryString, services, incomingAttachments);
        return executeQuery;
    }

    static MemoryStream BuildStream()
    {
        var stream = new MemoryStream();
        var writer = new StreamWriter(stream);
        writer.Write("Incoming");
        writer.Flush();
        stream.Position = 0;
        return stream;
    }

    public Tests(ITestOutputHelper output) : base(output)
    {
    }
}