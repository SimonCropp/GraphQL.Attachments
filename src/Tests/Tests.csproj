<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.2</TargetFramework>
    <Optimize>false</Optimize>
    <DebugType>full</DebugType>
    <DebugSymbols>true</DebugSymbols>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="GraphQL" Version="2.4.0" />
    <PackageReference Include="ApprovalTests" Version="3.0.20" />
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.2.0" />
    <PackageReference Include="ObjectApproval" Version="6.0.0" />
    <PackageReference Include="Xunit" Version="2.4.1" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1" />
    <PackageReference Include="MarkdownSnippets" Version="9.0.2" />
    <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
    <PackageReference Include="Microsoft.AspNetCore.Http" Version="2.2.0" />
    <ProjectReference Include="..\GraphQL.Attachments\GraphQL.Attachments.csproj" />
    <ProjectReference Include="..\Shared\Shared.csproj" />
  </ItemGroup>

</Project>