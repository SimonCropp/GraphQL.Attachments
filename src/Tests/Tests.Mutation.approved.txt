{
  ExecutionResult: {
    data: {
      withAttachment: {
        argument: 'argumentValue'
      }
    }
  },
  Attachments: {
    HasPendingAttachments: true,
    Inner: {
      key: {
        Headers: null,
        Value: 'Incoming'
      }
    }
  }
}